src/main/java/de/tum/in/www1/artemis/web/rest/TextExerciseResource.java

```diff
@@ -349,6 +349,11 @@ public ResponseEntity<StudentParticipation> getDataForTextEditor(@PathVariable L
             // set reference to participation to null, since we are already inside a participation
             textSubmission.setParticipation(null);
 
+            if (ExerciseDateService.isBeforeAssessmentDueDate(textExercise)) {
+                textSubmission.setResults(Collections.emptyList());
```

Comment:
Are you sure that immutability is of no concern here?

Reply:
There are no automatic results for text exercises currently, so no results should be sent to the student before the assessment due date.